engine/src/flutter/impeller/renderer/backend/vulkan/texture_source_vk.h

```diff
@@ -159,9 +158,7 @@ class TextureSourceVK {
  private:
   SharedHandleVK<vk::Framebuffer> framebuffer_;
   SharedHandleVK<vk::RenderPass> render_pass_;
-  mutable RWMutex layout_mutex_;
```

Comment:
This mutex is incorrect. The layout depends on code submitted to the queue, not the order of encoded operations. so layout_ is only a best guess and isn't correct if there is shared texture usage across more than one thread.

We don't do that today so there aren't any drawbacks to the bug.